core/src/language/python.rs

```diff
@@ -621,6 +458,318 @@ impl Python {
         };
         Ok(())
     }
+
+    fn gen_unit_variant_ctor(
```

Comment:
Also for the three constructor functions, theres some duplication that we can clean up like the method name and even the writing of the functions